src/app/submission/sections/upload/file/file.component.ts

```diff
@@ -0,0 +1,194 @@
+import { FileService } from '../../../../core/shared/file.service';
+import { HALEndpointService } from '../../../../core/shared/hal-endpoint.service';
+import { SubmissionJsonPatchOperationsService } from '../../../../core/submission/submission-json-patch-operations.service';
+import { SubmissionObject } from '../../../../core/submission/models/submission-object.model';
+import { WorkspaceitemSectionUploadObject } from '../../../../core/submission/models/workspaceitem-section-upload.model';
+
+@Component({
+  selector: 'ds-submission-upload-section-file',
+  styleUrls: ['./file.component.scss'],
+  templateUrl: './file.component.html',
+})
+export class UploadSectionFileComponent implements OnChanges, OnInit {
+
+  @Input() availableAccessConditionOptions: any[];
+  @Input() availableAccessConditionGroups: Map<string, any>;
+  @Input() collectionId;
+  @Input() collectionPolicyType;
+  @Input() configMetadataForm: SubmissionFormsModel;
+  @Input() fileId;
+  @Input() fileIndex;
+  @Input() fileName;
+  @Input() sectionId;
+  @Input() submissionId;
+
+  public fileData: WorkspaceitemSectionUploadFileObject;
+  public formId;
+  public readMode;
+  public formModel: DynamicFormControlModel[];
+
+  protected pathCombiner: JsonPatchOperationPathCombiner;
+  protected subscriptions = [];
+
+  constructor(private cdr: ChangeDetectorRef,
+              private fileService: FileService,
+              private formService: FormService,
+              private halService: HALEndpointService,
+              private modalService: NgbModal,
+              private operationsBuilder: JsonPatchOperationsBuilder,
+              private operationsService: SubmissionJsonPatchOperationsService,
+              private submissionService: SubmissionService,
+              private uploadService: SectionUploadService) {
+    this.readMode = true;
+  }
+
+  ngOnChanges() {
+    if (this.availableAccessConditionOptions && this.availableAccessConditionGroups) {
+      // Retrieve file state
+      this.subscriptions.push(
+        this.uploadService
+          .getFileData(this.submissionId, this.sectionId, this.fileId).pipe(
+          filter((bitstream) => isNotUndefined(bitstream)))
+          .subscribe((bitstream) => {
+              this.fileData = bitstream;
+            }
+          )
+      );
+    }
+  }
+
+  ngOnInit() {
+    this.formId = this.formService.getUniqueId(this.fileId);
+    this.pathCombiner = new JsonPatchOperationPathCombiner('sections', this.sectionId, 'files', this.fileIndex);
+  }
+
+  protected deleteFile() {
+    this.uploadService.removeUploadedFile(this.submissionId, this.sectionId, this.fileId);
+    this.operationsBuilder.remove(this.pathCombiner.getPath());
+    this.operationsService.jsonPatchByResourceID(
+      this.submissionService.getSubmissionObjectLinkName(),
+      this.submissionId,
+      this.pathCombiner.rootElement,
+      this.pathCombiner.subRootElement)
+      .subscribe();
```

Comment:
This subscribe is never unsubscribed. It also doesn't have any contents, does it need to be there?

Reply:
fixed